src/OpenSage.Game/Logic/Object/Update/UpdateModule.cs

```diff
@@ -34,13 +41,28 @@ internal virtual void Update(BehaviorUpdateContext context)
 
         SetNextUpdateFrame(context.LogicFrame + FramesBetweenUpdates);
         RunUpdate(context);
+
+        var sleepTime = Update();
+        SetNextUpdateFrame(context.LogicFrame + sleepTime.FrameSpan);
+    }
+
+    // TODO: Remove other Update methods after everything uses this.
```

Comment:
I've seen this a couple places now. Should the "other" methods be marked as `Obsolete` perhaps?

Reply:
I'm reluctant to do so just because it would add warnings. I'm expecting we'll gradually change everything to use the "correct" `Update` method as we port each `UpdateModule` across.